bulk_reassign_docs.py

```diff
@@ -0,0 +1,155 @@
+                moreResults = False
+            else:
+                pageID = str(resp['nextPage'])
+                nextPage = f'&page={pageID}'
+    return data
+
+def update_file(client, user_id, file_id):
+    """ Update file
+
+        :access_token:  Generated access token
+        :userId:        ID of the new owner
+        :file_id:       File to transfer ownership of
+
+        :returns:       Response JSON
+
+    """
+    try:
+        response = client.patch(
+            f"v2/files/{file_id}",
+            json={ "ownerId":user_id }
+        )
+        response.raise_for_status()
+    # HTTP and other errors are handled generally by raising them as exceptions to be surfaced in downstream code
+    except requests.exceptions.HTTPError as errh:
+        # The API response's message value is sent in lieu of the full response to display to the user for clarity
+        # Certain error response messages are useful for the user troubleshoot common issues, such as invalid Member Type or New Email already in use
+        raise Exception(errh.response.status_code, f"API message: {errh.response.json()['message']}")
+    except requests.exceptions.ConnectionError as errc:
+        raise Exception(f"Connection Error: {errc}, API response: {errc.response.text}")
+    except requests.exceptions.Timeout as errt:
+        raise Exception(f"Timeout Error: {errt}, API response: {errt.response.text}")
+    except requests.exceptions.RequestException as err:
+        raise Exception(f"Other Error: {err}, API response: {err.response.text}")
+    else:
+        data = response.json()
+        print("transfer of document:",file_id, "---", response )
+        return data
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Transfer all a users documents to another user')
+    parser.add_argument(
+        '--env', type=str, required=True, help='env to use: [production | staging].')
+    parser.add_argument(
+        '--cloud', type=str, required=True, help='Cloud to use: [aws | gcp | azure]')
+    parser.add_argument(
+        '--client_id', type=str, required=True, help='Client ID generated from within Sigma')
+    parser.add_argument(
+        '--client_secret', type=str, required=True, help='Client secret API token generated from within Sigma')
+    parser.add_argument(
+        '--curr_owner', type=str, required=True, help='Org Member who currently owns the documents')
+    parser.add_argument(
+        '--new_owner', type=str, required=True, help='Org Member who you want to transfer the documents to')
+
+    args = parser.parse_args()
+    client = SigmaClient(args.env, args.cloud, args.client_id, args.client_secret)
+    
+    # we need to confirm that both the existing user and new user are valid "check_users" fn
+    try:
+        curr_owner_id = get_member_id(client, args.curr_owner)
+    except Exception as e:
+        print(f"{e}")
+        raise SystemExit("Script aborted")
+    try:
```

Comment:
It might be a good idea to put a newline between each Try Except block for readability

Reply:
good call. I added the tabs and also used the VS code formatter which i had at some point turned off